### PR TITLE
Week2_day3_HW_solution

### PR DESCRIPTION
**Pull request by AzzamALjuaid**

### CODE REVIEW

File: src/Shape.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+open class Shape(val shapeName: String, val shapeColor: String) {
+    /*------------------------------------------------------------------------------*/
+        /***** functions called "info" that return name and color of shape.*****/
+    fun info() {
+        println("the shape name is:$shapeName and the color is:$shapeColor")
+    }
+    /*------------------------------------------------------------------------------*/
+        /***** functions called "computeArea" that has no parameter, and return Double data type.*****/
+    open fun computeArea(): Double {
+        return 0.0
```
Comment (shuruqt1000) [inline src/Shape.kt]: why is it open ? what does it mean ?

File: src/Shape.kt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        /***** subclass class called "Rectangle" that return the area of Rectangle.*****/
+class Rectangle(var width: Double, var height: Double, shapeName: String, shapeColor: String) :
+    Shape(shapeName, shapeColor) {
+    override fun computeArea(): Double = height * width
+}
+    /*------------------------------------------------------------------------------*/
+        /***** subclass class called "Circle" that return the area of Circle.*****/
+class Circle(var radius: Double, shapeName: String, shapeColor: String) : Shape(shapeName, shapeColor) {
+    val pi: Double = 3.14
+    override fun computeArea(): Double = pi * (radius * radius)
```
Comment (shuruqt1000) [inline src/Shape.kt]: what dose override mean and what exactly is being changed ?

File: src/Shape.kt [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+class Triangle(var base: Double, var height: Double, shapeName: String, shapeColor: String) :
+    Shape(shapeName, shapeColor) {
+    override fun computeArea(): Double = (height * base) / 2
+}
+    /*------------------------------------------------------------------------------*/
+        /***** subclass class called "Rectangle" that return the area of Rectangle.*****/
+class Rectangle(var width: Double, var height: Double, shapeName: String, shapeColor: String) :
+    Shape(shapeName, shapeColor) {
+    override fun computeArea(): Double = height * width
+}
```
Comment (shuruqt1000) [inline src/Shape.kt]: what dose these classes do and where do they get values from 
why is it important to have an override function there

**Review by shuruqt1000 — CHANGES_REQUESTED** [review]

File: src/Shape.kt (hunk 1 above)
Comment (AzzamALjuaid) [reply src/Shape.kt]: it is open to allow subclass (Triangle,Rectangle,Circle) to change in body of the functions and compute the area according to the subclass' name.

File: src/Shape.kt (hunk 2 above)
Comment (AzzamALjuaid) [reply src/Shape.kt]: we write "override" allow us to change the body of the functions we inherit from "computeArea()" and the functions we inherit from "computeArea()" most be open

File: src/Shape.kt (hunk 3 above)
Comment (AzzamALjuaid) [reply src/Shape.kt]: we "override" fun computeArea()" in each subclass to make sure this function will compute the area according to the subclass' name.
-we get values from main functions when we're testing every subclass

**Comment by AzzamALjuaid** [comment]
i reply in all comment

**Comment by shuruqt1000** [comment]
great comments  
completed

**Pull request closed without merging**